### PR TITLE
#SA7F-2

### PR DESCRIPTION
**Pull request by Tubka**

### CODE REVIEW

File: src/models/income.models.ts [hunk 1]
```diff
@@ -26,8 +26,12 @@ export interface IncomesWithImportDate extends IncomesWithYear {
   importDate: string | null;
 }
 export interface IncomesDb extends IncomesWithImportDate {
-  id: string | null;
+  id: string | number;
 }
```
Comment (hendrysiak) [inline src/models/income.models.ts:30]: First - if we have unused interface (or another code lines), we can delete it.  
Second - Do we use id in number type?

File: src/pages/Team/MembersTable/MembersTable.js [hunk 2]
```diff
@@ -1,38 +1,53 @@
-import React from 'react'
```
Comment (hendrysiak) [inline src/pages/Team/MembersTable/MembersTable.js:1]: We should delete unused components to improve app code

File: src/pages/Team/Team.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    { field: 'title', headerName: 'TYTUŁ', width: 800, },
+    { field: 'importDate', headerName: 'DATA', width: 150, },
+  ];
+
+  useEffect(() => {
+    const filteredIncomes = rows && rows.filter(i => {
+      if (useDate && selectedDate && i.importDate !== selectedDate.toLocaleString().split(',')[0]) return false;
+      if (event !== '' && i.event !== event && event !== 'unAssigned') return false;
+      if (event !== '' && event !== 'unAssigned' && !(i.name && i.surname && i.event && i.importDate && i.team && i.title && i.year && i.cash)) return false;
+      if (event !== '' && event === 'unAssigned' && i.name && i.surname && i.event && i.importDate && i.team && i.title && i.year && i.cash) return false;
```
Comment (hendrysiak) [inline src/pages/Team/Team.tsx:88]: Very long code line we should split to clearly version  
  
DO:

if (a

&& b

&& c)

DON'T:  
if(a && b && c)

Eslint guards the line lenght - we should always resolve all eslint's problems.

File: src/shared/Form/Form.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import Modal from '@material-ui/core/Modal';
```
Comment (hendrysiak) [inline src/shared/Form/Form.tsx:1]: To consider - if we use Form components only in Team.tsx, we can move this file to /pages/Team/components.  
  
In /shared we should hold only really "sharable" components (e.g Table Editor, maybe DatePicker).

File: src/shared/Form/Form.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    <>
+      <MailIcon onClick={handleOpen} style={{fontSize: 26, color: 'white', cursor: 'pointer'}}/>
+      <Modal
+        open={isOpen}
+        onClose={handleClose}
+        aria-labelledby="simple-modal-title"
+        aria-describedby="simple-modal-description"
+      >
+        <>
+          <form className={`${classes.positionModal}`}>
```
Comment (hendrysiak) [inline src/shared/Form/Form.tsx:28]: Why we don't use MUI Inputs? The styles of this inputs are not the same and we have UI conflicts.

File: src/shared/TeamFinanses/TeamFinanses.tsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
+import Modal from '@material-ui/core/Modal';
```
Comment (hendrysiak) [inline src/shared/TeamFinanses/TeamFinanses.tsx:1]: Next file potentially to move

File: src/shared/TeamFinanses/TeamFinanses.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import AttachMoneyIcon from '@material-ui/icons/AttachMoney';
+import React, {useEffect } from 'react';
+
+import axios from 'axios-income';
+
+import classes from './TeamFinanses.module.css';
+
+interface IMember {
```
Comment (hendrysiak) [inline src/shared/TeamFinanses/TeamFinanses.tsx:17]: Do we use this interface?

File: src/shared/TeamFinanses/TeamFinanses.tsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+});
+
+const TeamFinanses = ({ incomes, currentTeam } : Props): JSX.Element => {
+
+  const [onePercent, setOnePercent] = React.useState<number>(0);
+  const classe = useStyles();
+
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+
+  function createData(name: string, estate: number, expenses: number) {
```
Comment (hendrysiak) [inline src/shared/TeamFinanses/TeamFinanses.tsx:47]: Why not arrow function?

File: src/shared/TeamFinanses/TeamFinanses.tsx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+  useEffect(() => {
+    const getData = async () => {
+      const result = await axios.get(`/onePercent/${currentTeam}.json`);
+      setOnePercent(result.data);
+    };
+    getData();
+  });
+
+  const sum = (incomes ? incomes / 5 : 0) + onePercent;
```
Comment (hendrysiak) [inline src/shared/TeamFinanses/TeamFinanses.tsx:64]: In future - we should consider using useMemo or useCallback (  
if the calculations are demanding)

File: src/shared/TeamPage/TeamPage.tsx [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  useEffect(() => {
+    const helper = members ? (members.map((el, index) => {
```
Comment (hendrysiak) [inline src/shared/TeamPage/TeamPage.tsx:39]: "helper" is not correct name for this variable. We use setRows, so we can expect these to be "rows"

Comment (hendrysiak) [inline src/shared/TeamPage/TeamPage.tsx:39]: "helper" is not correct name for this variable. We use setRows, so we can expect these to be "rows"

**Review by hendrysiak — COMMENTED** [review]
Some comments to consider ;)

Generally - code is even clean and of a good level, but we should always do our best ;)

File: src/shared/Form/Form.tsx (hunk 4 above)
Comment (Tubka) [reply src/shared/Form/Form.tsx:1]: It's good idea, i created folder components and moved Form there.

File: src/shared/TeamFinanses/TeamFinanses.tsx (hunk 6 above)
Comment (Tubka) [reply src/shared/TeamFinanses/TeamFinanses.tsx:1]: Done.

File: src/shared/TeamFinanses/TeamFinanses.tsx (hunk 8 above)
Comment (Tubka) [reply src/shared/TeamFinanses/TeamFinanses.tsx:47]: i copied this from DOCS. I Changed to arrow function

**Pull request merged**